Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/boosterinfodetails/BoosterInfoDetailsViewModel.kt

```diff
@@ -0,0 +1,55 @@
+package de.rki.coronawarnapp.covidcertificate.boosterinfodetails
+
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.ccl.dccwalletinfo.model.BoosterNotification
+import de.rki.coronawarnapp.ccl.dccwalletinfo.text.textResource
+import de.rki.coronawarnapp.ccl.dccwalletinfo.text.urlResource
+import de.rki.coronawarnapp.covidcertificate.person.core.PersonCertificatesProvider
+import de.rki.coronawarnapp.util.coroutine.AppScope
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.mapNotNull
+import timber.log.Timber
+
+class BoosterInfoDetailsViewModel @AssistedInject constructor(
+    dispatcherProvider: DispatcherProvider,
+    private val personCertificatesProvider: PersonCertificatesProvider,
+    @AppScope private val appScope: CoroutineScope,
+    @Assisted private val personIdentifierCode: String
+) : CWAViewModel(dispatcherProvider) {
+
+    private val uiStateFlow = personCertificatesProvider.personCertificates.mapNotNull { certificateSet ->
+        UiState(
+            certificateSet.first { it.personIdentifier?.codeSHA256 == personIdentifierCode }
+                .dccWalletInfoWrapper.dccWalletInfo.boosterNotification
+        )
+    }.catch { error ->
+        Timber.d(error, "No person found for $personIdentifierCode")
```

Comment:
What should happen in this case? Should we return back or display some fallback strings?

Reply:
On its way to clarify with UX and Max 👀

Reply:
I think you can navigate back and in the person details screen the tile should be gone

Reply:
> 



> I think you can navigate back and in the person details screen the tile should be gone

yes, otherwise user will stay on the screen with broken UI